nordic nrf uart driver fails to build for multiple targets/configurations

**Describe the bug**
The nordic nrf uart driver fails to build for multiple targets/configurations
At least for this targets:
* nrf52840dk/nrf52840
* nrf5340_audio_dk/nrf5340/cpuapp
* nrf54h20dk/nrf54h20/cpuapp
* nrf52_bsim

**To Reproduce**
Steps to reproduce the behavior:
For ex.:
1. mkdir build; cd build
2. cmake -GNinja -DBOARD=nrf52840dk/nrf52840  ../tests/drivers/uart/uart_async_api 
3. ninja
4. See error

**Expected behavior**
No build failures

**Impact**
Main CI failing, many PRs CI failing 

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/runs/29794121010
```
drivers/serial/uart_nrfx_uarte.c: In function 'rxto_isr':
drivers/serial/uart_nrfx_uarte.c:518:46: error: 'struct uarte_nrfx_data' has no member named 'flags'
  518 |         (IS_ENABLED(UARTE_HW_ASYNC) ? (config->flags & UARTE_CFG_FLAG_HW_BYTE_COUNTING) : false)
      |                                              ^~
drivers/serial/uart_nrfx_uarte.c:1309:21: note: in expansion of macro 'HW_RX_COUNTING_ENABLED'
 1309 |                 if (HW_RX_COUNTING_ENABLED(data)) {
      |                     ^~~~~~~~~~~~~~~~~~~~~~
```

**Environment (please complete the following information):**

 - OS: Linux, Ubuntu 22.04 CI
 - Toolchain (e.g Zephyr SDK, ...): 0.16.8
 - Commit SHA or Version used: main as of now a80af336c23f447ad259c492ec8322133a0e3cbf

**Additional context**
Introduced by https://github.com/zephyrproject-rtos/zephyr/pull/72476

--- a/drivers/serial/uart_nrfx_uarte.c
+++ b/drivers/serial/uart_nrfx_uarte.c
@@ -1306,7 +1306,7 @@ static void rxto_isr(const struct device *dev)
 
 		data->async->discard_rx_fifo = false;
 		flushed = rx_flush(dev, NULL, 0);
-		if (HW_RX_COUNTING_ENABLED(data)) {
+		if (HW_RX_COUNTING_ENABLED(config)) {
 			/* It need to be included because TIMER+PPI got RXDRDY events
 			 * and counted those flushed bytes.
 			 */

